Add support for CITEXT type

arteam opened the pull request:
I would like to contribute `CITEXT` support, requested in #188.

As far as I understand, it could be implement lust as a simple string decoder/encoder. So it should be a fairly simple change.

guylabs commented:
@arteam Thanks for the contribution I just wanted to start on it next week :wink:

jesperpedersen commented:
Thanks for your contribution.

However, the pull request is failing checkstyle checks (`mvn clean package`), and you need to disable the tests if the CITEXT extension isn't installed - see PreparedStatementTest and hstore for an example.

Commit:
Add support for CITEXT type

`CITEXT` is a case-insensitive character string type supported by
PostgreSQL through an extension.

arteam commented:
Thanks for your pointers. I fixed Checkstyle warnings and added JUnit assumptions about existence of `CITEXT`. Let me know if it's good enough. 

P.S. For some reason Travis CI doesn't want to run the build.

jesperpedersen commented:
Looks ok to me now.

@kdubb Ok to merge ?

kdubb commented:
Looks good!
